docs/content/preview/contribute/core-database/merge-with-upstream-repositories.md

```diff
@@ -0,0 +1,1017 @@
+1. If backporting to stable branches,
+
+   - ...if the PG version is the same, the previously constructed subtree commit can be reused for [subtree merge](#git-subtrees).
+   - ...if the PG version is different and a single multi-PG-version compatible `yb` branch is used in the upstream repository, same thing.
+   - ...otherwise, the same process should be repeated using the `yb-pg<version>` branch corresponding to that stable branch's PG version.
+
+#### Subtree direct-descendant merge
+
+The steps that follow will use documentdb as a hypothetical example.
+Suppose that YugabyteDB was based off a yugabyte/documentdb repo `yb` branch constructed from `v0.102-0` with commit [b896737f53d9eb13e0b397976e1b2edd310cca57](https://github.com/microsoft/documentdb/commit/b896737f53d9eb13e0b397976e1b2edd310cca57) cherry-picked.
+We are trying to merge to `v0.103-0` (which doesn't exist at the time of writing, but you get the idea of the example).
+
+1. First, determine whether a merge in the upstream repository can be skipped.
+   This is only permissible if the current upstream repository referenced commit is not a YugabyteDB commit (e.g. there is no `yb` branch since a `microsoft/documentdb` commit is used directly).
+   Then, the target version can directly be used.
+   (In this example, this shortcut can not be used.)
+1. Otherwise, switch to a feature branch off the latest yugabyte/documentdb repo `yb` branch: `git switch -c merge-documentdb yb`.
+   Then, do `git merge v0.103-0`.
+   (It may still be the case that the target version, in this example `v0.103-0`, already has contains all the changes YB imported on top of `v0.102-0`.
+   If that is the case, then the merge can be simplified to take exactly the target version: `git merge -X theirs v0.103-0`.)
+   Do any relevant testing if applicable.
+   Record merge resolutions into the merge commit message.
+   There should only be the merge commit and no other extraneous commits on top of it.
+   Make a GitHub PR of this through your fork for review.
+1. On the [yugabyte/yugabyte-db][repo-yugabyte-db] repo, [subtree merge](#git-subtrees) this branch.
+
+#### Subtree non-direct-descendant merge
+
+The steps that follow will use pgaudit as an example.
+
+This merge can technically be handled similarly as [direct-descendant merge](#squash-direct-descendant-merge).
+However, there is a much faster alternative that should be taken.
+A prerequisite for using this alternative is that the target version contains all the commits of the source version (or, nearly equivalently, the target version was released later than the source version).
+For example, `1.7.0` should contain all commits in `1.3.2`, considering backports as equivalent to each other, and for any commits only in PG `1.3.2`, they are likely irrelevant for PG `1.7.0`.
+This prerequisite generally shouldn't fail since major version upgrades should aim for the latest minor version of a major version.
```

Comment:
Since this paragraph is very similar to the other paragraph describing when a non-direct-descendant merge can be done, should we move them into one joint section called "When is a non-direct-descendant merge possible?" to avoid the paragraphs diverging?